### ISSUE TITLE
Will the key in MapAccess support Express?

### ISSUE THREAD

**Issue by monadbobo** [apache/datafusion-sqlparser-rs#380]
The index of an array in ClickHouse can be an function such as :

```sql
string.values[indexOf(string.names, 'endpoint')]='/health'
```

PS: If sqlparser wants to support this syntax,I can submit a PR to support it.

**Comment by houqp** [comment]
Looks like a reasonable feature to add? ping @alamb @Dandandan @Jimexist

**Comment by alamb** [comment]
I agree extending support for MapAccess sounds like a good idea to me 👍

**Comment by LucaCappelletti94** [comment]
Closing as completed in https://github.com/apache/datafusion-sqlparser-rs/pull/382
